capstone-frontend/src/components/ClubPage.jsx

```diff
@@ -117,6 +124,16 @@ class ClubPage extends Component {
         .catch(e => console.log(e));
   }
 
+  deleteMeeting = (meetingID) => {
+    let meetings = this.state.meetings;
+    let meetingIDsArray = this.state.meetings.map(m => m.id);
+    const index = meetingIDsArray.indexOf(meetingID);
+    if (index > -1) {
+      meetings.splice(index, 1);
```

Comment:
This is actually mutating your component's state, since `meetings` is just a reference to `this.state.meetings` (not a copy). You can fix this by using the `filter` method instead of `splice` (since `splice` alters an array in-place and `filter` returns a new array) or by changing the declaration of `meetings` above to `const meetings = [...this.state.meetings];`

Reply:
I didn't realize that! My intention was to avoid manipulating the state directly, since I know that is advised against. I'll change it to use the spread operator like you mentioned so that `meetings` holds a copy of the state, rather than a reference to the state itself. Thanks for the feedback!